src/pystack/_pystack/process.cpp

```diff
@@ -360,7 +359,14 @@ AbstractProcessManager::getStringFromAddress(remote_addr_t addr) const
         len = unicode._base._base.length;
         buffer.resize(len);
 
-        data_addr = ((remote_addr_t)((char*)addr + sizeof(PyASCIIObject)));
+        size_t offset;
+        if (d_major > 3 || (d_major == 3 && d_minor >= 12)) {
+            offset = sizeof(Python3_12::PyASCIIObject);
+        } else {
+            offset = sizeof(Python3::PyASCIIObject);
+        }
```

Comment:
Maybe we can create some kind of structure that holds these types when a version is selected so we don't need to branch out in the code like this. What do you think?

Reply:
I'm not happy with this, but I couldn't spot a nicer way to write it that didn't seem to be adding a _lot_ of complexity. I started by defining the full structure of a `PyUnicodeObject` for each version, but I decided it was overkill and threw it away when I realized that it adds a fair amount of code, and that there's nothing we actually care about beyond the size of the `PyASCIIObject`, at least as things stand today. If you can see a nicer way to do this one, feel free to add a commit.

Reply:
I tried something but is a lot of code and quite complex 😞

Let's leave it like this and hopefully we can find some better way to do this.

Reply:
An idea for the future is having sone struct of sizes as well as one of offsets.

Reply:
> I tried something but is a lot of code and quite complex 😞

Yeah, I'll be honest: what you see in this PR was my 3rd attempt at this, and it's the least ugly one 😄. And the second least ugly had a magic `sizeof(wchar_t*)` in it, heh. I'm sure this could be improved, but I don't think it's easy, and I think this isn't too bad. I think my vote is to leave this alone until we have the next new `PyUnicodeObject` layout, and decide at that point how best to factor out the commonalities across them.